onadata/apps/logger/tests/models/test_xform.py

```diff
@@ -167,6 +168,26 @@ def test_check_xform_uuid(self):
         except DuplicateUUIDError as e:
             self.fail("DuplicateUUIDError raised: %s" % e)
 
+    @mock.patch('onadata.apps.viewer.models.data_dictionary.set_uuid')
+    def test_gen_xform_uuid(self, set_uuid_mock):
+        """
+        Test that an XForms uuid is always set by default on new XForm
+        Objects
+        """
+        self._publish_transportation_form()
+        uuid = self.xform.uuid
+        # The _publish_transportation_form uses the DataDictionary
+        # Model which has some processing steps that ensure XForms
+        # Always have uuids.
+        # Assert that set_uuid is called with an object
+        # that already has a uuid.
+        self.assertEqual(set_uuid_mock.call_count, 1)
+        self.assertEqual(set_uuid_mock.call_args[0][0].uuid, uuid)
+        # Assert that the uuid is not empty and is
+        # 32 characters in length
+        self.assertNotEqual(uuid, '')
+        self.assertEqual(len(uuid), 32)
```

Comment:
uuid is always 36 characters (32 hex digits + 4 dashes). Using the default way uuids are set for `DataDictionary` objects; which only returns the hex values.